server/channels/api4/apitestlib.go

```diff
@@ -796,6 +804,23 @@ func (th *TestHelper) CreateDmChannel(user *model.User) *model.Channel {
 	return channel
 }
 
+func (th *TestHelper) CreateThreadPostWithClient(client *model.Client4, channel *model.Channel, rootPost *model.Post) *model.Post {
```

Comment:
nit: I'm hesitant adding global test helpers that are used in just a single test. Would we consider keeping these in channel_test.go for now until we expect to share it more widely?

Reply:
Totally agree, and maybe I should have made a companion ticket about it already. In the user's thread for the current team API tests, we need something like this as well, and I'm planning to update this there as well, just not as part of this ticket.

For example in these cases:
https://github.com/mattermost/mattermost-server/blob/3552f4dccbbf8cb5700fec67043d80a371aff2ee/server/channels/api4/user_test.go#L6126

https://github.com/mattermost/mattermost-server/blob/3552f4dccbbf8cb5700fec67043d80a371aff2ee/server/channels/api4/user_test.go#L6130

https://github.com/mattermost/mattermost-server/blob/3552f4dccbbf8cb5700fec67043d80a371aff2ee/server/channels/api4/user_test.go#L6156